framework/src/outputs/AdvancedOutputUtils.C

```diff
@@ -18,31 +18,34 @@
 #include "InputParameters.h"
 
 // Constructor of OutputOnWarehouse; initializes the MultiMooseEnums for all available output types
-OutputOnWarehouse::OutputOnWarehouse(const MultiMooseEnum & execute_on,
+OutputOnWarehouse::OutputOnWarehouse(const ExecFlagEnum & execute_on,
                                      const InputParameters & parameters)
-  : OutputMapWrapper<MultiMooseEnum>()
+  : OutputMapWrapper<ExecFlagEnum>()
 {
   // Initialize each of the 'execute_on' settings for the various types of outputs
-  if (parameters.have_parameter<MultiMooseEnum>("execute_nodal_on"))
+  if (parameters.have_parameter<ExecFlagEnum>("execute_nodal_on"))
     _map.insert(std::make_pair("nodal", execute_on));
 
-  if (parameters.have_parameter<MultiMooseEnum>("execute_elemental_on"))
+  if (parameters.have_parameter<ExecFlagEnum>("execute_elemental_on"))
     _map.insert(std::make_pair("elemental", execute_on));
 
-  if (parameters.have_parameter<MultiMooseEnum>("execute_scalars_on"))
+  if (parameters.have_parameter<ExecFlagEnum>("execute_scalars_on"))
     _map.insert(std::make_pair("scalars", execute_on));
 
-  if (parameters.have_parameter<MultiMooseEnum>("execute_postprocessors_on"))
+  if (parameters.have_parameter<ExecFlagEnum>("execute_postprocessors_on"))
     _map.insert(std::make_pair("postprocessors", execute_on));
 
-  if (parameters.have_parameter<MultiMooseEnum>("execute_vector_postprocessors_on"))
+  if (parameters.have_parameter<ExecFlagEnum>("execute_vector_postprocessors_on"))
     _map.insert(std::make_pair("vector_postprocessors", execute_on));
 
-  if (parameters.have_parameter<MultiMooseEnum>("execute_input_on"))
-    _map.insert(std::make_pair("input", Output::getExecuteOptions()));
+  if (parameters.have_parameter<ExecFlagEnum>("execute_input_on"))
+    _map.insert(std::make_pair("input", MooseUtils::getDefaultExecFlagEnum()));
 
-  if (parameters.have_parameter<MultiMooseEnum>("execute_system_information_on"))
-    _map.insert(std::make_pair("system_information", Output::getExecuteOptions("initial")));
+  if (parameters.have_parameter<ExecFlagEnum>("execute_system_information_on"))
+  {
+    _map.insert(std::make_pair("system_information", MooseUtils::getDefaultExecFlagEnum()));
+    _map["system_information"] = EXEC_INITIAL;
```

Comment:
Wait, two inserts for the same key? Maybe you have an overload so that the option just gets changed but it looks screwy. I'd prefer you build the right value and insert once.

Reply:
Not two inserts, the _map.insert creates the entry. The second line sets the current flag to EXEC_INITIAL, using the operator= from the MultiMooseEnum class.

Reply:
OK - so you are using the overloaded operator here. I think it would be much clearer to just build the value you need, then insert it.

Reply:
I can do that.